pydoc/developer/BENCHMARK.md

```diff
@@ -0,0 +1,470 @@
+bench_decode[ascii]     3.4621 (1.04)     4.9177 (1.11)     4.2250 (1.13)     0.6125 (1.50)     4.0197 (1.17)     0.9991 (1.90)          2;0  236.6864 (0.88)          5           1
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+```
+Similarly, to only run a single test within a file, specify the test name with the `-k` flag instead of a filename
+```commandline
+python3 -m pytest -c benchmark.ini --benchmark-autosave --benchmark-storage=file://benchmark_v2/.benchmarks -k bench_encode
+```
+Results:
+```commandline
+benchmark_v2/encoding_benchmark.py ..                                                                                                                                         [100%]
+Saved benchmark data in: <Arkouda_root>/benchmark_v2/.benchmarks/Linux-CPython-3.9-64bit/0015_31de39be8b19c76d073a8999def6673a305c250d_20230405_145947_uncommited-changes.json
+
+-------------------------------------------------------------------- benchmark 'Strings_EncodeDecode': 2 tests ---------------------------------------------------------------------
+Name (time in ms)          Min               Max              Mean            StdDev            Median               IQR            Outliers       OPS            Rounds  Iterations
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+bench_encode[ascii]     3.4298 (1.0)      3.6450 (1.0)      3.5541 (1.0)      0.0889 (1.0)      3.5801 (1.00)     0.1436 (1.0)           2;0  281.3620 (1.0)           5           1
+bench_encode[idna]      3.4875 (1.02)     4.5255 (1.24)     3.7912 (1.07)     0.4328 (4.87)     3.5652 (1.0)      0.4869 (3.39)          1;0  263.7659 (0.94)          5           1
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+```
+
+## Benchmark Arguments
+There are a large number of commandline arguments available for configuring the benchmarks to run in a way fitting 
+to any use case.
+
+<table>
+<tr><th>Argument</th><th>Default Value</th></tr>
+<tr><th colspan="2">Description</th></tr>
+<tr>
+    <td>--benchmark-autosave</td>
+    <td>True when using "<b>make benchmark"</b></td>
+</tr>
+<tr><td colspan="2">Save the benchmark JSON results to the provided storage location</td></tr>
+<tr>
+    <td>--benchmark-storage</td>
+    <td>"file://benchmark_v2/.benchmarks" when using "<b>make benchmark"</b></td>
+</tr>
+<tr><td colspan="2">Storage location to save benchmark output JSON</td></tr>
+<tr>
+    <td>--benchmark-save</td>
+    <td>"COMMIT-ID_DATE_TIME_IS-DIRTY" example: 
+31de39be8b19c76d073a8999def6673a305c250d_20230405_145947_uncommited-changes.json</td>

```

Comment:
Probably want one that is clean here because most users won't have this occur.

Reply:
Updated